arch/riscv/core/stacktrace.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2024 Meta Platforms
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/kernel_structs.h>
+#include <kernel_internal.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
+
+uintptr_t z_riscv_get_sp_before_exc(const z_arch_esf_t *esf);
+
+#if __riscv_xlen == 32
+ #define PR_REG "%08" PRIxPTR
+#elif __riscv_xlen == 64
+ #define PR_REG "%016" PRIxPTR
+#endif
```

Comment:
I would guess that `%p` would prefix with `0x` and and zero-pad values appropriately, but then they would need to be cast to `(void *)` as well.

Maybe the following would be better? It skips the `0x` prefix though (but at least there is no cast required. 
```suggestion
#if __riscv_xlen == 32
 #define PR_REG "%08" PRIx32
#elif __riscv_xlen == 64
 #define PR_REG "%016" PRIx64
#endif
```

Reply:
Will take a look there when I do the symtab PR after this

Reply:
Sure - this isn't really blocking though, so if it works, it works.

Reply:
The current implementation works without casting, `PRIx64`/`PRIx32` have to be casted